brainiak/fcma/fcma.py

```diff
@@ -0,0 +1,346 @@
+
+        Parameters
+        ----------
+        task: tuple (start_voxel_id, num_assigned_voxels)
+            depicting the voxels assigned to compute
+        corr: 3D array in shape [num_selected_voxels, num_epochs, num_voxels]
+            the normalized correlation values of all subjects in all epochs
+            for the assigned values, in row-major
+
+        Returns
+        -------
+        results: list of tuple (voxel_id, accuracy)
+            the accuracy numbers of all voxels, in accuracy descending order
+            the length of array equals the number of assigned voxels
+        """
+        (sv, e, av) = corr.shape
+        kernel_matrix = np.zeros((e, e), np.float32, order='C')
+        results = []
+        for i in range(sv):
+            if isinstance(clf, sklearn.svm.SVC) \
+                    and clf.kernel == 'precomputed':
+                blas.compute_kernel_matrix('L', 'T',
+                                           e, self.num_voxels,
+                                           1.0, corr,
+                                           i, self.num_voxels,
+                                           0.0, kernel_matrix, e)
+                data = kernel_matrix
+            else:
+                data = corr[i, :, :]
+            # no shuffling in cv
+            skf = cross_validation.StratifiedKFold(self.labels,
+                                                   n_folds=self.num_folds,
+                                                   shuffle=False)
+            scores = cross_validation.cross_val_score(clf, data, self.labels,
+                                                      cv=skf, n_jobs=1)
+            results.append((i + task[0], scores.mean()))
+        return results
+
+    def voxelScoring(self, task, clf):
+        """ voxel selection processing done in the worker node
+
+        Take the task in,
+        do analysis on voxels specified by the task (voxel id, num_voxels)
+        It is a three-stage pipeline consisting of:
+        1. correlation computation
+        2. within-subject normalization
+        3. voxelwise cross validaion
+
+        Parameters
+        ----------
+        task: tuple (start_voxel_id, num_assigned_voxels),
+            depicting the voxels assigned to compute
+        clf: classification function
+            the classifier to be used in cross validation
+
+        Returns
+        -------
+        results: list of tuple (voxel_id, accuracy)
+            the accuracy numbers of all voxels, in accuracy descending order
+            the length of array equals the number of assigned voxels
+        """
+        time1 = time.time()
+        # correlation computation
+        # corr is a 3D array in row major,
+        # in (selected_voxels, epochs, all_voxels) shape
+        # corr[i, e, s + j] = corr[j, e, s + i]
+        corr = self.correlationComputation(task)
+        # normalization
+        # in-place z-score, the result is still in corr
```

Comment:
The functionality of `correlationNormalization`/`normalization` is already documented in their docstrings.